File: vue-netcore-chatroom/Models/RegisterPasswordUserRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace vue_netcore_chatroom.Models
{
    public class RegisterPasswordUserRequest
    {
        [Required]
        [EmailAddress]
        [MaxLength(255)]
        public string Email { get; set; }

        [MaxLength(255)]
        public string? FirstName { get; set; }

        [MaxLength(255)]
        public string? LastName { get; set; }

        [Required]
        [MinLength(8)]
        public string Password { get; set; }

        [Required]
        [Compare("Password")]
        public string ConfirmPassword { get; set; }

    }
}

File: client/src/models/RegisterPasswordUserRequest.ts

interface IRegisterPasswordUserRequest {
  email: string;
  password: string;
  confirmPassword: string;
  firstName: string | null;
  lastName: string | null
}

export default class RegisterPasswordUserRequest implements IRegisterPasswordUserRequest {
  email: string = "";
  password: string = "";
  confirmPassword: string = "";
  firstName: string | null = null;
  lastName: string | null = null;

  // constructor(data: IRegisterPasswordUserRequest ) {
  //   this.email = data.email;
  //   this.password = data.password;
  //   this.confirmPassword = data.confirmPassword;
  //   this.firstName = data.firstName;
  //   this.lastName = data.lastName;
  // }
}